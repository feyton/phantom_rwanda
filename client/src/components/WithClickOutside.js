import React, { useCallback, useEffect, useRef, useState } from 'react';
import { useLocation } from 'react-router-dom';

const WithClickOutside = (WrappedComponent) => {
  const Component = () => {
    const [open, setOpen] = useState(false);
    const location = useLocation();

    const ref = useRef(null);

    const clickOutside = () => {
      const handleClickOutside = (event) => {
        if (ref.current && !ref.current.contains(event.target)) {
          setOpen(false);
        }
      };
      document.addEventListener('mousedown', handleClickOutside);
    };

    const setRef = useCallback(() => {
      clickOutside();
      return [ref];
    }, []);

    useEffect(() => {
      open && setOpen(false);
    }, [location.key]);
    const [newRef] = setRef();

    return <WrappedComponent open={open} setOpen={setOpen} ref={newRef} />;
  };

  return Component;
};

export default WithClickOutside;
