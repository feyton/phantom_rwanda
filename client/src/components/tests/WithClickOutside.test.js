import React from 'react';
import renderer from 'react-test-renderer';
import WithClickOutside from '../WithClickOutside.js';

describe('withClickOutside', () => {
  it('should render withClickOutside', () => {
    const withClickOutside = renderer
      .create(
        <div>
          <WithClickOutside />
        </div>
      )
      .toJSON();
    expect(withClickOutside).toMatchSnapshot();
  });
});
