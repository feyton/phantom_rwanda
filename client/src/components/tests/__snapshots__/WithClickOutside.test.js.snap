// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withClickOutside should render withClickOutside 1`] = `<div />`;
